tests/middleware_tests/test_fieldkeys.py

```diff
@@ -0,0 +1,113 @@
+from bibtexparser.middlewares.enclosing import RemoveEnclosingMiddleware
+from bibtexparser.middlewares.fieldkeys import NormalizeFieldKeys
+from bibtexparser.splitter import Splitter
+
+test_bibtex_string_lowercasekeys = """
+@article{smith2022,
+  author  = "Smith, J.",
+  title   = "A Test Article",
+  journal = "J. of Testing",
+  month   = "jan",
+  year    = "2022"
+}
+
+@book{doe2021,
+  author    = "Doe, J.",
+  title     = "A Test Book",
+  publisher = "Test Pub.",
+  year      = "2021",
+  month     = apr
+}
+
+@inproceedings{jones2023,
+  author    = "Jones, R.",
+  title     = "A Test Conf. Paper",
+  booktitle = "Proc. of the Intl. Test Conf.",
+  year      = "2023",
+  month     = 8
+}
+"""
+
+test_bibtex_string_capitalizedkeys = """
+@article{smith2022,
+  Author  = "Smith, J.",
+  Title   = "A Test Article",
+  Journal = "J. of Testing",
+  Month   = "jan",
+  Year    = "2022"
+}
+
+@book{doe2021,
+  Author    = "Doe, J.",
+  Title     = "A Test Book",
+  Publisher = "Test Pub.",
+  Year      = "2021",
+  Month     = apr
+}
+
+@inproceedings{jones2023,
+  Author    = "Jones, R.",
+  Title     = "A Test Conf. Paper",
+  Booktitle = "Proc. of the Intl. Test Conf.",
+  Year      = "2023",
+  Month     = 8
+}
+"""
+
+def test_normalize_lowercase():
```

Comment:
Isn't the easiest check to ensure that there are no changes to `test_bibtex_string_lowercasekeys` is `NormalizeFieldKeys` is applied?

Reply:
I basically modeled the tests after `test_month.py`, because I'm not that familiar with `bibtexparser` yet. :)

The original string is preprocessed through a `Splitter`, which returns a `Library`. Then feeding the library (not the original string!) through the middleware returns a new `Library`. Libraries do not have an `__eq__` method, so comparing them will compare the `id`, thus failing the equality test.

I suppose if we export both libraries back into a string (to normalize whitespace) and compare those strings, this approach would be possible. But that relies on the exporter working correctly (i.e. this test unit is then no longer independent).

What do you think, shall I change it?

Reply:
I would be as simple/short as possible: parse back and forth to a string. However, you do make a fair point. @MiWeiss do you have a preference?

Reply:
> I suppose if we export both libraries back into a string (to normalize whitespace) and compare those strings, this approach would be possible. But that relies on the exporter working correctly (i.e. this test unit is then no longer independent).

Exactly, I likewise don't like that too much for the same reason. I would refrain from using the writer in any unit tests not directly related to bibtex writing.

Note: For the same reason, you *could* have avoided starting with a BibTeX string, but directly constructed your test data as Entry-objects. That way, the test would be really atomic. But on this end, I do not worry enough to make you re-write everything :-)

That's just my five cents though, as I was explicitly asked for it :-) If you decide to go either way, I'll be ok with it. In the end, @tdegeus is the reviewer of this PR...

Reply:
No big preference. However, if you have the stamina, I would prefer the simplest based on `Entry`.

Reply:
I'll see what I can do. This test unit is not that many lines of code...

Reply:
Reimplemented tests using `Entry`.

This doesn't make them shorter, but at least makes the test unit more independent. :)